### PR TITLE
first cut at enhanved provenance and nb generator

### PR DESCRIPTION
**Pull request by doutriaux1**
moved recipes into repo as well
fixed version number in setup.py

@muryanto1 I'm sure I didn't setup coverage correctly do you mind taking a look to make sure it's ok.

Once this is in, I will push a PR for vcs to generate pngs properly using this generateProvenace, same for cdms

@zshaheen you should take a look at generateProvenance function and see if we can enhance it for docker.

### CODE REVIEW

**Comment by muryanto1** [comment]
I have updated the .circleci/config.yml to run coverage for linux py3, and also to run py3 only if py2 passes. Tests passed.

**Comment by doutriaux1** [comment]
@muryanto1 thanks!

**Comment by zshaheen** [comment]
@doutriaux1 When building a container image, an environmental variable is set in the `Dockerfile`. So when the software is running, we can check for this. If it's present and the correct value, you can set an `is_container` value in the provenance dictionary to `True` or something similar.

**Review by zshaheen — APPROVED** [review]

**Comment by doutriaux1** [comment]
@muryanto1 @dnadeau4 @zshaheen looking at coverage results from here, we do not get to see the details but it's reading the coverage from the python env not the source trre (see : https://coveralls.io/builds/19458002/source?filename=workspace/test_linux_cdat_info_py3/miniconda/envs/py3/lib/python3.6/site-packages/cdat_info/cdat_info_src.py) I think using @zshaheen approach, i.e NOT using `distutils` but `setuptools` will help. We are using it (poorly becuase we still need to use `--old-and-unmanageable`) for vcs and we can see the source code on PR see: (https://coveralls.io/builds/19182541/source?filename=vcs/vtk_ui/button.py)

What do you guys think? I think we should slowly convert all our packages to this method. I"m not 100% sure it is possible for the ones needing access to numpy array (such as cdms2 and genutil) but it should be ok.

**Comment by doutriaux1** [comment]
@dnadeau4 since you're in charge of provenance your review is very important. Feel free to edit/change/comment on this. Also I do know that the history is only working from jupyter lab/notebook right now. If you want to look into this that would be nice.

**Comment by doutriaux1** [comment]
> @doutriaux1 When building a container image, an environmental variable is set in the `Dockerfile`. 
So when the software is running, we can check for this. If it's present and the correct value, you can set an `is_container` value in the provenance dictionary to `True` or something similar.

@zshaheen I think we would need a bit more than that, just knowing we're in a container is not enough for provenance, we would need the container's info as well.

File: tests/test_cdat_info_generate_notebook.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
....19.1=py36_1\n  - send2trash=1.5.0=py_0\n  - setuptools=40.4.0=py36_1000\n  - simplegeneric=0.8.1=py_1\n  - sip=4.18.1=py36hfc679d8_0\n  - six=1.11.0=py36_1\n  - sortedcontainers=2.0.5=py_0\n  - sqlalchemy=1.2.12=py36h470a237_0\n  - sqlite=3.25.2=hb1c47c0_0\n  - tblib=1.3.2=py_1\n  - terminado=0.8.1=py36_1\n  - testpath=0.3.1=py36_1\n  - tk=8.6.8=ha92aebf_0\n  - toolz=0.9.0=py_0\n  - tornado=5.1.1=py36h470a237_0\n  - traitlets=4.3.2=py36_0\n  - udunits2=2.2.27.6=h3a4f0e9_1\n  - urllib3=1.23=py36_1\n  - wcwidth=0.1.7=py_1\n  - webencodings=0.5.1=py_1\n  - wheel=0.31.1=py36_1001\n  - widgetsnbextension=3.4.2=py36_0\n  - x264=1!152.20180717=h470a237_1\n  - xz=5.2.4=h470a237_1\n  - yaml=0.1.7=h470a237_1\n  - zeromq=4.2.5=hfc679d8_5\n  - zict=0.1.3=py_0\n  - zlib=1.2.11=h470a237_3\n  - blas=1.0=mkl\n  - intel-openmp=2019.0=118\n  - libcurl=7.61.0=hf30b1f0_0\n  - libgfortran=3.0.1=h93005f0_2\n  - mkl=2019.0=118\n  - numpy=1.15.1=py36h6a91979_0\n  - numpy-base=1.15.1=py36h8a80b8c_0\n  - pycurl=7.43.0.2=py36hdbc3d79_0\n  - pip:\n    - msgpack==0.5.6\n    - mv2==3.0.0\n    - regrid2==3.0.0\n    - sidecar==0.2.0\n    - testrunner==0.0.0\nprefix: /Users/doutriaux1/anaconda2/envs/widgets3\n\n'
```
Comment (dnadeau4) [inline tests/test_cdat_info_generate_notebook.py]: @doutriaux1 Is there another way to do this?  Can you put that in a file or do a `conda env export`?  Can easily break as is...

**Comment by dnadeau4** [comment]
I think we should forget about `docker` right now and push this true for v1.0.  It is very good as is...

**Comment by doutriaux1** [comment]
@dnadeau4 I completely agree docker stuff should be in a separate PR down the road. @dnadeau4 if you're happy with this, please approve. We still have the caveat though that history is not captured outside of jupyterlab at the moment but we can fix this in another branch

**Review by dnadeau4 — APPROVED** [review]

**Comment by doutriaux1** [comment]
@dnadeau4 once this is approved I can submit PRs for both vcs and cdms, as initial point, but I will let you take decide on the best way(s) to really implement this.

File: tests/test_cdat_info_generate_notebook.py (hunk 1 above)
Comment (doutriaux1) [reply tests/test_cdat_info_generate_notebook.py]: @dnadeau4 I just thought of a better way to do this. I'll push something soon

**Review by dnadeau4 — APPROVED** [review]

**Pull request merged**